meta/rpc.go

```diff
@@ -63,6 +65,8 @@ func (r *rpc) proxyLeader(conn *net.TCPConn) {
 	defer leaderConn.Close()
 
 	leaderConn.Write([]byte{MuxRPCHeader})
+	// re-write the original message to the leader
+	leaderConn.Write(buf)
```

Comment:
We used to always forward everything to the leader.  No, depending on the message, it may not be intended for the leader.  In the case of promoting a new raft node, we explicitly don't want to talk to the leader, but talk to the node that needs to be promoted to the raft cluster.  The read was refactored into two other methods to make it easier to interrogate the message coming in.  Then, if we did want to talk to the leader, we have to re-write the original message, which is what we are doing above.